src/modules/RA/DataLayer/components/useSourceData.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { useDataProvider } from 'react-admin';
+import { useField } from 'react-final-form';
+
+import { RES_DATASOURCE } from '../../ra-modules';
+
+/**
+ * Returns the source data
+ * @param {str} sourceFieldName
+ */
+export const useSourceData = sourceFieldName => {
+  const { input: { value: sourceId } } = useField(sourceFieldName);
+  const dataProvider = useDataProvider();
+
+  const [sourceData, setSourceData] = React.useState({});
+  const loadingRef = React.useRef(false);
+
+  React.useEffect(() => {
+    let isMounted = true;
+
+    const loadSourceData = async () => {
+      const { data } = await dataProvider.getOne(RES_DATASOURCE, { id: sourceId });
```

Comment:
Maybe could `dataProvider.getOne()` be replaced by [`useGetOne()`](https://marmelab.com/react-admin/Actions.html#specialized-hooks) specialized hook.

*(Tests should be made to check behaviors precisely)*

Reply:
I already tried but I can't do that, a provider can't be conditional ;-)